### PR TITLE
Update main code to be encapsulated in CLI and use click

### PR DESCRIPTION
**Pull request by cthoyt**
I'm not a big fan of global state - I'd also prefer to communicate this back and forth using the app's config rather than global state in the module, but I'll send an additional PR for this

### CODE REVIEW

File: curation_service/app.py [hunk 1]
```diff
@@ -286,7 +264,11 @@ def update_curations():
                  ('pa_hash', 'stmt_hash'), 'source_hash']
 
     # Build up the curation dict.
+    db_key = "primary"
     db = get_db('primary')
+    if db is None:
+        raise RuntimeError(f"unable to get database: {db_key}")
```
Comment (cthoyt) [inline curation_service/app.py:270]: this will raise if your configuration for the primary database is old  (like mine, ATM)

**Comment by bgyori** [comment]
This changes the behavior of the CLI, the way I used to run it doesn't work with this PR (I think one thing that changed is turning --tag into an optional argument from a positional one). This and #6 look okay but also not really necessary since the CLI and app have been working well so far.

**Comment by cthoyt** [comment]
@bgyori I can switch it to mimic the old functionality with click, but the change to automatically use the current working directory seems pretty helpful. I'll at least switch the other two back from options to arguments.

These two PRs are part of a push to package this code so it can be pip installed then the command can be run directly as `$ indra_curation` - this requires splitting up some of the code that uses global state into proper functions

**Comment by bgyori** [comment]
Okay, so I just wasted a bunch of time because I just noticed that this PR actually broke something (as I noted at the time this change wasn't really necessary and things were working fine before...). Fix coming soon.

**Pull request merged**